File: mssuaweb-catalogos/src/main/java/mx/gob/imss/catalogos/dto/RegistroPatronalDto.java
package mx.gob.imss.catalogos.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class RegistroPatronalDto {
    private String registroPatronal;
    private String razonSocial; // Nombre corregido para coincidir con tu Service
    private String rfc;
    private String domicilio;
}

File: guisuaweb_portal/src/app/business/catalogos/model/RegistroPatronalDto.ts
/**
 * DTO que representa un Registro Patronal vinculado a un RFC.
 * Utilizado para el catálogo con filtro de búsqueda en la Solicitud Patronal.
 */
export interface RegistroPatronalDto {
  // Identificador único del registro patronal (ej. A1234567810)
  registroPatronal: string;

  // Nombre o Razón Social del patrón
  razonSocial: string;

  // RFC vinculado al registro
  rfc?: string;

  // Opcionales: podrías incluir otros datos si el backend los devuelve
  cveIdDelegacion?: number;
  cveIdSubdelegacion?: number;
}
